plugin_specific_configuration.cpp

```diff
@@ -17,8 +17,7 @@ namespace irods {
                     const auto& inst_name = rule_engine.at( CFG_INSTANCE_NAME_KW ).get_ref<const std::string&>();
                     if ( inst_name == _instance_name ) {
                         if(rule_engine.count(CFG_PLUGIN_SPECIFIC_CONFIGURATION_KW) > 0) {
-                            return boost::any_cast<const std::unordered_map<std::string, boost::any>&>(
-                                    rule_engine.at(CFG_PLUGIN_SPECIFIC_CONFIGURATION_KW));
+                            return rule_engine.at(CFG_PLUGIN_SPECIFIC_CONFIGURATION_KW);
```

Comment:
this function looks like it is returning a copy instead of a const ref.

to not hold you up on this PR, consider capturing this in an issue.

Reply:
Done: irods/irods#6057